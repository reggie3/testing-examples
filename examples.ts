// This is a pure function.
// All information it needs in order to work is passed as
// an argument or available as part of JS
export const add = (a: number, b: number) => {
  return a + b;
};

export const double = (a: number) => {
  return a * 2;
};

export const asyncDouble = (a: number) => {
  return new Promise((resolve, reject) => {
    setTimeout(() => {
      resolve(double(a));
    }, 1000);
  });
};

export const getAddressOfWhiteHouse = () => {
  return {
    street: "1600 Pennsylvania Ave NW",
    city: "Washington",
    state: "DC",
    zip: "20500"
  };
};
