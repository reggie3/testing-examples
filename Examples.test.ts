import { add, double, asyncDouble, getAddressOfWhiteHouse } from "./examples";

// description block to help organize tests
describe("Basic Examples", () => {
  // Basic synchronous test
  it("should add correctly ", () => {
    expect(add(1, 1)).toEqual(2);
  });

  // Async test to test an async function
  it("should ", async () => {
    const result = await asyncDouble(4);
    expect(result).toEqual(8);
  });

  // "each" test
  // eash array takes arguments to be passed as inputs to
  // function under test with the expected result as the
  // last item in an array
  it.each([
    [1, 2],
    [2, 4]
  ])("double(%d)", (input, expected) => {
    expect(double(input)).toBe(expected);
  });

  //only run this test
  // it.only("should ", () => {});

  // skip this test
  it.skip("should ", () => {});

  // placeholder
  it.todo("should ");
});

describe("Snapshot Example", () => {
  it("should provide the correct address for the White House ", () => {
    expect(getAddressOfWhiteHouse()).toMatchSnapshot();
  });
  it("should provide the correct address for the White House (inline snapshot)", () => {
    expect(getAddressOfWhiteHouse()).toMatchInlineSnapshot(`
      Object {
        "city": "Washington",
        "state": "DC",
        "street": "1600 Pennsylvania Ave NW",
        "zip": "20500",
      }
    `);
  });
});
