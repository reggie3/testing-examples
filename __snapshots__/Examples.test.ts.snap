// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Example should provide the correct address for the White House  1`] = `
Object {
  "city": "Washington",
  "state": "DC",
  "street": "1600 Pennsylvania Ave NW",
  "zip": "20500",
}
`;
